src/System/Handleable.h

```diff
@@ -0,0 +1,44 @@
+#ifndef _HANDLEABLE_H
+#define _HANDLEABLE_H
+
+#include "Particle.h"
+#include "Handler.h"
+// Forward declaration due to mutual inclusion
```

Comment:
taking notes on this one

Reply:
Yes this is what I was trying to explain (badly) on Saturday. When you have 2 header files that include each other, and you use define guards to not have an infinite include loop, file A will include file B, but then file B can't include file A because of the define guards. Both the files end up getting compiled ultimately, but until that happens you need a way of telling one of them about the existence of a class in the other if you're going to use that class.

Reply:
It might be better to make the handleables a nested class since it doesn't really make sense without a handler anyway. Would remove the compile co-dependency and also make it a bit cleaner IMO.

Reply:
would it not make more sense to make the handler a nested class, since Handleable is the one that's being implemented in all the other classes?

Reply:
I would also think that Handler should be the nested class if we go that route. I hadn't thought about that, but it makes sense. Alternatively, the Handler functionality could be implemented in Handleable with some static members variables and methods.